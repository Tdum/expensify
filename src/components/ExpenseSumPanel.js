import getExpensesTotal from '../../selectors/expenses-total.js'
import expenses from '../tests/fixtures/expenses'

const total = getExpensesTotal(expenses)
console.log(total)