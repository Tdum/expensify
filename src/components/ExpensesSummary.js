import React from 'react'
import { connect } from 'react-redux'
import getExpensesTotal from '../selectors/expenses-total.js'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'


export const ExpensesSummary = ({ expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        expensesCount: selectExpenses(state.expenses, state.filters).length,
        expensesTotal: getExpensesTotal(selectExpenses(state.expenses, state.filters))
    }
}

export default connect(mapStateToProps)(ExpensesSummary)


// reprendre le mod√®le d'ExpenseList