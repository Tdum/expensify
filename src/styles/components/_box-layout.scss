.box-layout {
    align-items: center;
    background: $blue-pink-gradient;
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw
}

.box-layout__box {
    background: fade-out(white, .75);
    border-radius: 3px;
    padding: $l-size;
    text-align: center;
    width: 25rem;;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}