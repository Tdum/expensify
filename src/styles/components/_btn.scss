.btn {
    background-color: $dark-blue;
    border-radius: 3px;
    display: inline-block;
    color: white;
    font-size: $font-size-large;
    font-weight: 700;
    line-height: 1;
    padding: 8px 16px;
    border: none;
    text-decoration: none;
}

.btn--link {
    background: none;
    border: none;
    color: white;
    font-weight: 100;
}

.btn--secondary {
    background: #888888;
}