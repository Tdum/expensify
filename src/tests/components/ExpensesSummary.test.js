import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'


test('Should render ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={100.10}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should render ExpensesSummary with 2 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={2000.20}/>)
    expect(wrapper).toMatchSnapshot()
})