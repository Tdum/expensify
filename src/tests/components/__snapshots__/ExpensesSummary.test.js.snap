// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $1.00
</div>
`;

exports[`Should render ExpensesSummary with 2 expenses 1`] = `
<div>
  Viewing 
  2
   
  expenses
   totalling 
  $20.00
</div>
`;
